cmd/util/flags/cliflags/spec.go

```diff
@@ -57,13 +57,13 @@ func SpecFlags(settings *SpecFlagSettings) *pflag.FlagSet {
 		"i",
 		InputUsageMsg,
 	)
-	flags.StringSliceVarP(
+	flags.StringToStringVarP(
 		&settings.OutputVolumes,
 		"output",
 		"o",
 		settings.OutputVolumes,
-		`name:path of the output data volumes. `+
-			`'outputs:/outputs' is always added unless '/outputs' is mapped to a different name.`,
+		`name=path of the output data volumes. `+
+			`'outputs=/outputs' is always added unless '/outputs' is mapped to a different name.`,
```

Comment:
I don't think this is true anymore. Will need to address this before merging.

Reply:
why changed from `name:path` to `name=path`? this will break users. Also we will need to define a default output path `outputs:/outputs` for both `docker run` and `wasm run` to avoid breaking existing behavour and breaking all our examples and docs. I don't think it is a bad design anyways for those commands, but worth exploring the right long term option with @aronchick

Reply:
For context here @aronchick: This comment is about a change in behavior of the output flag from expecting `:` as a separator to `=` as a separator.

When using a flag that requires pairs of strings (key,value) - for example environment variables - and in the case here ResultPaths I find a `StringToStringVar` flag type (a map) to provide clearer validation compared to accepting a `StringArrayVar` flag (a slice). As an example, consider:

**Valid**:
`bacalhau docker run -o outputs=/outputs -o inputs=/inputs`

**Invalid**:
`bacalhau docker run -o outputs=/outputs -o inputs`

Cobra, our CLI library, will immediately catch the invalid case and return an error since `inputs` is missing a value field.

Comparatively, the behavior prior to this change accepted both options, and it was on us (developers) to write the validation logic to catch the invalid case of a key missing a value.

If we feel this is too large of a change for our users to understand - replacing `:` with `=` - then I can write a bespoke implementation of a map flag parser that uses `:` as the separator instead of `=` to preserve existing behavior.

Reply:
Can we please stop changing behaviour or adding more complexity to the migration story? The migration of the APIs and CLIs is a huge and complex story as is, and we've discussed that no change in behaviour to be done while we migrate to avoid as many issues as possible.

While I appreciate you identifying areas of improvements as you work on this task, the right way to handle this is by cutting issues to track them.